src/application/common/logger.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as Logger from 'bunyan';
+
+const ILogger = Symbol("ILogger");
+
+interface ILogger {
```

Comment:
What about the following design:
- a minimal API for the logger server
```ts
export namespace LogLevel {
  export const ERROR = 2;
  export const WARN = 1;
  export const INFO = 0;
}

export interface ILoggerServer {
  logLevel(): Promise<number>;
  log(logLevel: number, message: string, ...params: any[]): void;
}
```
- a logger service using the logger server that defines a convenient API and do `level` checking
```ts
export interface ILogger {
   error(message: string, params: ...any[]);
   warn(message: string, params: ...any[]);
   info(message: string, params: ...any[]);
}

export class Logger implements ILogger {

  protected readonly logLevel: Promise<number>;

  constructor(
    @inject(ILoggerServer) protected readonly server: ILoggerServer
  ) {
    this.logLevel = server.logLevel();  
  }

  protected log(logLevel: number, message: string, ...params: any[]): void {
    this.logLevel().then(level => {
      if (level <= logLevel) {
         this.server.log(logLevel, message, ...params)
      }
    });
  }

  error(message: string, ...params: any[]): void {
    this.log(LogLevel.ERROR, message, params);
  }

  info...
}
```
- a bunyan implementation of `ILoggerServer`: `BunyanLoggerServer`

I think it has a clean separation and easier to grasp compared to the proxy logic in `loggerServerModule` below. Also, we should always remember that `ILoggerServer` is special, we cannot just send a random js object with the JSON-RPC, e.g. maybe we should remove `...params` from the interface since it has any type. Others thoughts?